test/unit/providers/oauth-provider.spec.js

```diff
@@ -137,6 +158,94 @@ describe('OAuthProvider', function() {
       OAuthToken.removeToken();
     }));
 
+    describe('authorize()', function() {
+      var data = {
+        client_id: defaults.clientId,
+        response_type: 'code',
+        scope: 'foo:bar',
+        state: 'state_hash'
+      };
+
+      it('should throw an error if `clientId` is missing', inject(function(OAuth) {
+        try {
+          OAuth.authorize();
+
+          should.fail();
+        } catch(e) {
+          e.should.be.an.instanceOf(Error);
+          e.message.should.match(/clientId/);
+        }
+      }));
+
+      it('should call `queryString.stringify`', inject(function(OAuth) {
+        sinon.spy(queryString, 'stringify');
+
+        OAuth.authorize(data.client_id, data.scope, data.state);
+
+        queryString.stringify.callCount.should.equal(1);
+        queryString.stringify.firstCall.args.should.have.lengthOf(1);
+        queryString.stringify.firstCall.args[0].should.eql({
+          client_id: data.client_id,
+          response_type: 'code',
+          scope: data.scope,
+          state: data.state
+        });
+
+        queryString.stringify.restore();
+      }));
+
+      it('should return an error if request response doesn\'t contain a `redirectUri` attribute', inject(function($httpBackend, OAuth) {
```

Comment:
@joaogranado I had the same doubt when developing. So on your opinion, I should only test that `$http.get` should be called with proper data and leave that way?